tests_pytest/common_tests/unit_tests/core/quantization/quantization_params_generation/test_calculate_quantization_params.py

```diff
@@ -0,0 +1,118 @@
+from model_compression_toolkit.target_platform_capabilities import OpQuantizationConfig
+from model_compression_toolkit.core import QuantizationConfig
+from model_compression_toolkit.target_platform_capabilities.schema.mct_current_schema import Signedness
+from model_compression_toolkit.core.common.collectors.statistics_collector import StatsCollector
+from mct_quantizers import QuantizationMethod
+from model_compression_toolkit.core.common.node_prior_info import NodePriorInfo
+
+
+class TestCalculateQuantizationParams:
+    def build_op_cfg(self):
+        op_cfg = Mock(spec=OpQuantizationConfig)
+        op_cfg.activation_quantization_method = QuantizationMethod.POWER_OF_TWO
+        op_cfg.activation_n_bits = 16
+        op_cfg.enable_activation_quantization = True
+        op_cfg.quantization_preserving = False
+        op_cfg.signedness = Signedness.AUTO
+
+        return op_cfg
+
+    def build_node(self, name='node', q_mode=ActivationQuantizationMode.QUANT):
+        node = Mock(spec=BaseNode)
+        node.name = name
+        node.get_node_weights_attributes.return_value = []
+
+        if q_mode == ActivationQuantizationMode.QUANT:
+            node.is_activation_quantization_enabled.return_value = True
+            node.is_fln_quantization.return_value = False
+        elif q_mode == ActivationQuantizationMode.FLN_QUANT:
+            node.is_activation_quantization_enabled.return_value = False
+            node.is_fln_quantization.return_value = True
+        else:
+            node.is_activation_quantization_enabled.return_value = False
+            node.is_fln_quantization.return_value = False
+
+        activation_quantization_cfg = NodeActivationQuantizationConfig(op_cfg=self.build_op_cfg())
+        activation_quantization_cfg.quant_mode = q_mode
+
+        candidate_quantization_config = Mock(spec=CandidateNodeQuantizationConfig)
+        candidate_quantization_config.activation_quantization_cfg = activation_quantization_cfg
+        candidate_quantization_config.weights_quantization_cfg = Mock(spec=NodeWeightsQuantizationConfig)
+
+        node.candidates_quantization_cfg = [candidate_quantization_config]
+
+        return node
+
+    def get_test_graph(self, node_name, q_mode, data):
+        node = self.build_node(node_name, q_mode=q_mode)
+        graph = Graph('graph_name', input_nodes=[node], nodes=[node], output_nodes=[node], edge_list=[])
+
+        graph.node_to_out_stats_collector = dict()
+        for n in graph.nodes():
+            n.prior_info = NodePriorInfo()
+
+            graph.node_to_out_stats_collector[n] = StatsCollector(init_min_value=0.0, init_max_value=1.0, out_channel_axis=0)
+            graph.node_to_out_stats_collector[n].hc._n_bins = 3
+            graph.node_to_out_stats_collector[n].hc._bins = np.array(data)
+            graph.node_to_out_stats_collector[n].hc._counts = np.array([1, 1])
+
+        quant_config = QuantizationConfig()
```

Comment:
if this is the same (default QuantizationConfig) for all test cases, it is clearer to just initialized it in the test and not as part of the graph setup